### PR TITLE
WIP: Next major version

### PR DESCRIPTION
**Pull request by anajavi**
**Breaking changes:**

- [X] Drop immutable, lodash.isEqual
- [ ] Make Series & friends PureComponent
- [ ] Drop Highcharts < 7
- [ ] Drop React < 16.2
- [ ] Remove FlagSeries as it is now called FlagsSeries
- [ ] Drop support for IE < 11
  - [ ] babel-preset-env target IE = 11
- [ ] replace lodash with native functions where convenient
  - [X] forEach
  - [X] isFunction
  - [ ] pickBy
  - [ ] omitBy
  - [X] omit (to be removed from [lodash](https://github.com/lodash/lodash/wiki/Roadmap) )
  - [ ] attempt
  - [ ] find (likely removed from [lodash](https://github.com/lodash/lodash/commit/4ba0f8b6124062836af257dae8d208976b0e05b7#comments)
  - [ ] findKey
  - [ ] defaultTo
  - [ ] lowerFirst
  - [ ] mapKeys
  - [X] size
  - [ ] wrap (to be removed from [lodash](https://github.com/lodash/lodash/wiki/Roadmap) )
  - [ ] isPlainObject
  - [ ] isEmpty
  - [ ] findIndex
  - [ ] upperFirst

**Features:**
- [ ] Annotation component
- [ ] colorAxis support
- [ ] better responsiveness (yet to be defined)

**Other:**
- [ ] Update examples
- [ ] Update docs

As discussed on #171

### CODE REVIEW

File: packages/react-jsx-highcharts/src/components/Series/Series.js [hunk 1]
```diff
@@ -47,10 +43,7 @@ class Series extends Component {
 
     let needsRedraw = false;
     // Using setData is more performant than update
-    if (isImmutable(data) && immutableEqual(data, prevProps.data) === false) {
-      this.series.setData(data.toJS(), false);
```
Comment (whawker) [inline packages/react-jsx-highcharts/src/components/Series/Series.js]: I feel uneasy about dropping support for Immutable, I can't see how these few lines effect the bundle size that much, and Immutable is quite likely to be used in a React app.

Comment (anajavi) [reply packages/react-jsx-highcharts/src/components/Series/Series.js]: It's not about bundle size, it's only a couple of kB. I just thought that making equality checks is unnecessary. 

I can bring it back with immutable 4?

What about isEqual checks? I find it more inline with other libraries when doing just shallow compare?

Comment (anajavi) [reply packages/react-jsx-highcharts/src/components/Series/Series.js]: Actually if we keep immutable and isEqual, there is no need to make major release if Highcharts 7 does not break things.

All the new features and lodash changes can be made within version 3.

Might make more sense actually, as most React libraries seem to target their breaking changes to React 16.7.

I am not against that all. What do you think?

**Comment by anajavi** [comment]
Postponed.

**Pull request closed without merging**